build-tools/packages/build-cli/src/config.ts

```diff
@@ -63,8 +63,21 @@ export interface FlubConfig {
 	 * Configuration for the `generate:releaseNotes` command.
 	 */
 	releaseNotes?: ReleaseNotesConfig;
+
+	/**
+	 * The multiple of minor versions to use for calculating the next version in the legacy compatibility range.
+	 * This interval applies exclusively to the client release group; for all other release groups, the caret versions are used.
+	 */
+	legacyCompatInterval?: LegacyCompatInterval;
 }
 
+/**
+ * Represents legacy compat interval per release groups.
+ */
+export type LegacyCompatInterval = {
+	[releaseGroup in ReleaseGroup]: number;
```

Comment:
```suggestion
	[releaseGroupName: string]: number;
```

Better to use a string here, because there's nothing on the config side preventing you from entering any value you want in the config. So better to represent that in this type by making the key a string. That will force users of the config value to check that it's valid or to assume it's valid and blow up if not.

Reply:
Actually `Record<string, number>` is probably better.

Reply:
can you elaborate - `there's nothing on the config side preventing you from entering any value you want in the config`?

Reply:
Yeah, what I mean is that in the fluidBuild/flub config file, which is either JSON or CJS, I could write:

```js
const config = {
  legacyCompatInterval: {
    DoesNotExist: 10,
  },
};
```

when that config is loaded, it's cast to the type in TS - there's no TS compiler in the flow that enforces that the only valid values for the keys are the ones we know about. So if you type this as a release group, then code that reads the config will get a string value that is not in the string literal union of the type - that is, code is going to see "DoesNotExist" and treat it like a release group name.